sql-plugin/src/main/java/com/nvidia/spark/rapids/SpillableKudoTable.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.nvidia.spark.rapids;
+
+import ai.rapids.cudf.HostMemoryBuffer;
+import com.nvidia.spark.rapids.jni.kudo.KudoTable;
+import com.nvidia.spark.rapids.jni.kudo.KudoTableHeader;
+
+public class SpillableKudoTable extends KudoTable {
+  private SpillableHostBuffer shb;
```

Comment:
really this should just be a `SpillableHostBufferHandle`, but for now it probably fits better with the code.